src/Cli/dotnet/TransactionalAction.cs

```diff
@@ -65,19 +72,44 @@ public void Rollback(Enlistment enlistment)
             // This automatically inherits any ambient transaction
             // If a transaction is inherited, completing this scope will be a no-op
             T result = default(T);
-            using (var scope = new TransactionScope(
-                TransactionScopeOption.Required,
-                TimeSpan.Zero))
+            try
             {
-                Transaction.Current.EnlistVolatile(
-                    new EnlistmentNotification(commit, rollback),
-                    EnlistmentOptions.None);
+                using (var scope = new TransactionScope(
+                    TransactionScopeOption.Required,
+                    TimeSpan.Zero))
```

Comment:
Are you sure you want an infinite timeout? A really high timeout will still eventually timeout whereas something going wrong with an infinite timeout can cause things to block indefinitely.

Reply:
I think that the other operations happening within the transaction have their own timeouts, so it wouldn't end up being effectively infinite.